src/background_thread.c

```diff
@@ -364,7 +364,11 @@ background_thread_create(tsd_t *tsd, unsigned arena_ind) {
 	pre_reentrancy(tsd);
 	int err;
 	load_pthread_create_fptr();
-	if ((err = pthread_create(&info->thread, NULL,
+	/*
+	 * To avoid complications (besides reentrancy), create internal
```

Comment:
After having done some refactoring around this code, it's not obvious to me what the complications are that we're trying to avoid.  What problems did you hit that motivated this change?